### PR TITLE
[BPF] Allow selectively disable compiler builtins for BPF target

### PR DESCRIPTION
**Pull request by dmakarov**

### CODE REVIEW

File: llvm/lib/Analysis/TargetLibraryInfo.cpp [hunk 1]
```diff
@@ -1503,9 +1509,9 @@ bool TargetLibraryInfoImpl::getLibFunc(const Function &FDecl,
                                        LibFunc &F) const {
   // Intrinsics don't overlap w/libcalls; if our module has a large number of
   // intrinsics, this ends up being an interesting compile time win since we
-  // avoid string normalization and comparison.
```
Comment (jackcmay) [inline llvm/lib/Analysis/TargetLibraryInfo.cpp:1506]: These changes probably not necessary

File: llvm/lib/Analysis/TargetLibraryInfo.cpp [hunk 2]
```diff
@@ -550,6 +550,12 @@ static void initialize(TargetLibraryInfoImpl &TLI, const Triple &T,
     TLI.setUnavailable(LibFunc_nvvm_reflect);
   }
 
+  if (T.isBPF()) {
+    TLI.setUnavailable(LibFunc_rust_alloc);
```
Comment (jackcmay) [inline llvm/lib/Analysis/TargetLibraryInfo.cpp:554]: Why does llvm have to have any knowledge of these rust functions?

Comment (dmakarov) [reply llvm/lib/Analysis/TargetLibraryInfo.cpp:554]: it does. It replaces them with internal built-ins, which these calls to `TLI.setUnavailable()` prevent from happening.

File: llvm/lib/Analysis/TargetLibraryInfo.cpp (hunk 1 above)
Comment (dmakarov) [reply llvm/lib/Analysis/TargetLibraryInfo.cpp:1506]: let's keep it... the trailing white space is bad in source code (my editor automatically deletes it on saving a file)

**Pull request merged**